lib/sentimental.rb

```diff
@@ -1,24 +1,26 @@
 class Sentimental
-  attr_accessor :threshold, :word_scores, :neutral_regexps, :ngrams
+  attr_accessor :threshold, :word_scores, :neutral_regexps, :ngrams, :influencers
 
-  def initialize(threshold: 0, word_scores: nil, neutral_regexps: [], ngrams: 1)
-    if ngrams >= 1
-      @ngrams = ngrams.to_i
-    else
-      @ngrams = 1
-    end
+  def initialize(threshold: 0, word_scores: nil, neutral_regexps: [], ngrams: 1, influencers: nil)
+    @ngrams = ngrams >= 1 ? ngrams.to_i : 1
     @word_scores = word_scores || {}
+    @influencers = influencers || {}
     @word_scores.default = 0.0
+    @influencers.default = 0.0
     @threshold = threshold
     @neutral_regexps = neutral_regexps
   end
 
   def score(string)
-    return 0 if neutral_regexps.any? {|regexp| string =~ regexp}
+    return 0 if neutral_regexps.any? { |regexp| string =~ regexp }
 
-    extract_words_with_n_grams(string).inject(0) do |score, token| 
+    @influence = 0.0
+    @total_score = extract_words_with_n_grams(string).inject(0) do |score, token|
+      @influence += influencers[token]
       score += word_scores[token]
     end
+
+    @total_score + influence_score
```

Comment:
That is a good point. I will take a look at changing it to a multiplier based off of the next